### PR TITLE
Add a performance tip

### PR DESCRIPTION
**Pull request by putianyi889**

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -57,6 +57,18 @@ A[band(0)] .= -2/h^2
 A[band(1)] .= A[band(-1)] .= 1/h^2
 ```
 
+Creating a large banded matrix from a dense matrix should be avoided because that costs time and memory:
+```jldoctest
+julia> @time BandedMatrix(ones(10000,10000),(0,0));
+0.775120 seconds (10 allocations: 763.016 MiB, 28.04% gc time)
+```
+Try to use abstract arrays to get around this:
```
Comment (dlfivefifty) [inline docs/src/index.md]: "abstract array" is not the right term here. you could say FillArrays.jl

Comment (putianyi889) [reply docs/src/index.md]: FillArrays cannot do everything. I added a new example showing this.

Comment (dlfivefifty) [reply docs/src/index.md]: The issue is that the arrays are not "abstract": they are concrete though non-standard. Perhaps we can say "structured matrices"?

Comment (putianyi889) [reply docs/src/index.md]: Done.

**Pull request merged**